### PR TITLE
Add contracts for System.Version

### PR DESCRIPTION
**Pull request by yaakov-h**
Cross-checked with Reference Source.

I've verified that this compiles, and work as expected for .NET 4.5.

### CODE REVIEW

**Comment by hubuk** [comment]
It should be safe to also add the following contracts for properties.
```C#
//for Major and Minor
Contract.Ensures(Contract.Result<int>() >= 0);
//for Build and Revision
Contract.Ensures(Contract.Result<int>() >= -1);
//for MajorRevision and MinorRevision
Contract.Ensures(Contract.Result<short>() >= -1);
```

**Comment by yaakov-h** [comment]
Thanks @hubuk.

Confirmed still compiling.

File: Microsoft.Research/Contracts/MsCorlib/System.Version.cs [hunk 1]
```diff
@@ -22,24 +22,58 @@ namespace System
...
         {
-            get { return default(int); }
+            get {
+              Contract.Ensures(Contract.Result<int>() >= -1);
+              return default(int);
+            }
+        }
+
+        [Pure]
+        public int MajorRevision
```
Comment (hubuk) [inline Microsoft.Research/Contracts/MsCorlib/System.Version.cs]: I haven't spotted this before. [Version.MajorRevision](https://msdn.microsoft.com/en-us/library/system.version.majorrevision%28v=vs.110%29.aspx) is of short type.

File: Microsoft.Research/Contracts/MsCorlib/System.Version.cs [hunk 2]
```diff
@@ -22,24 +22,58 @@ namespace System
...
+        {
+            get {
+              Contract.Ensures(Contract.Result<int>() >= -1);
+              return default(int);
+            }
+        }
+
+        [Pure]
+        public int MinorRevision
```
Comment (hubuk) [inline Microsoft.Research/Contracts/MsCorlib/System.Version.cs]: Same for [Version.MinorRevision](https://msdn.microsoft.com/en-us/library/system.version.minorrevision%28v=vs.110%29.aspx).

File: Microsoft.Research/Contracts/MsCorlib/System.Version.cs (hunk 1 above)
Comment (yaakov-h) [inline Microsoft.Research/Contracts/MsCorlib/System.Version.cs]: Good catch, I'll fix them up tomorrow morning.

**Pull request merged**